File: LeetCode/Easy/0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        int MARK = 100_000;
        while (head != null) {
            if (head.val == MARK) {
                return true;
            }
            head.val = MARK;
            head = head.next;
        }
        return false;
    }
}

File: LeetCode/Easy/0141-linked-list-cycle/0141-linked-list-cycle.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
    ptr1 := head
    ptr2 := head

    for (ptr1 != nil && ptr2 != nil) {
        ptr1 = ptr1.Next
        ptr2 = ptr2.Next

        if (ptr2 == nil) {
            break
        }

        ptr2 = ptr2.Next

        if (ptr1 == ptr2) {
            return true
        }
    }
    return false;
}